src/main/java/org/terracotta/statistics/derived/EventParameterSimpleMovingAverage.java

```diff
@@ -190,8 +173,8 @@ private void archive(AveragePartition partition) {
 
     private final LongAdder total = new LongAdder();
     private final LongAdder count = new LongAdder();
-    private final LongMaxUpdater maximum = new LongMaxUpdater();
-    private final LongMaxUpdater minimum = new LongMaxUpdater();
+    private final LongAccumulator maximum = new LongAccumulator((x, y) -> x > y ? x : y, Long.MIN_VALUE);
```

Comment:
We can use `Math::max` and `Math::min` here.

Reply:
Indeed